web/src/engine/websites/MangaLib.ts

```diff
@@ -1,46 +1,125 @@
+import { RateLimit } from '../taskpool/RateLimit';
 
-function MangaExtractor(anchor: HTMLAnchorElement) {
-    return {
-        id: anchor.pathname,
-        title: anchor.querySelector('.media-card__title').textContent.trim()
-    };
-}
-
-const chapterScript = `
-    new Promise(resolve => {
-        const chapters = window.__DATA__.chapters.list.map(entry => {
-            return {
-                id: '/' + window.__DATA__.manga.slug + '/v' + entry.chapter_volume + '/c' + entry.chapter_number,
-                title: entry.chapter_number + (entry.chapter_name ? ' - ' + entry.chapter_name : '')
-            };
-        });
-        resolve(chapters);
-    });
-`;
-
-const pageScript = `
-    new Promise(resolve => {
-        resolve(window.__pg.map(page => window.__info.servers.main + window.__info.img.url + page.u));
-    });
-`;
-
-@Common.MangaCSS(/^{origin}\/[^/]+(\?section=info)?$/, 'div.media-name__body div.media-name__main')
-@Common.MangasMultiPageCSS('/manga-list?page={page}', 'div.media-card-wrap a.media-card', 1, 1, 0, MangaExtractor)
-@Common.ChaptersSinglePageJS(chapterScript, 500)
-@Common.PagesSinglePageJS(pageScript, 500)
-@Common.ImageAjax()
+type ImageServers = {
+    imageServers: ImageServer[]
+}
+
+type ImageServer = {
+    id: string,
+    url: string,
+    site_ids: number[]
+}
+
+type APIResult<T> = {
+    data: T,
+    meta: {
+        has_next_page: boolean,
+    }
+}
+
+type APIManga = {
+    id: number,
+    rus_name: string,
+    slug_url: string
+}
+
+type APIChapter = {
+    volume: string,
+    number: string,
+    name: string,
+    branches_count: number,
+    branches: {
+        id: number,
+        branch_id: number | null,
+        teams: {
+            name: string
+        }[]
+    }[]
+}
+
+type APIPages = {
+    pages: {
+        url: string
+    }[]
+}
+
+type ChapterID = {
+    branch_id: string,
+    number: string,
+    volume: string
+}
+
+@Common.ImageAjax(true)
 export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://api.mangalib.me/api/';
+    private readonly siteId = 1;
+    private imageServer: ImageServer = undefined;
+    private readonly mangasTaskPool = new TaskPool(1, new RateLimit(2, 1));
 
     public constructor() {
-        super('mangalib', 'MangaLib', 'https://mangalib.org', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Russian, Tags.Source.Aggregator);
+        super('mangalib', 'MangaLib', 'https://mangalib.org', Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Media.Manga, Tags.Language.Russian, Tags.Source.Aggregator);
     }
 
     public override get Icon() {
         return icon;
     }
 
+    public override async Initialize(): Promise<void> {
+        const { data: { imageServers } } = await FetchJSON<APIResult<ImageServers>>(new Request(new URL('constants?fields[]=imageServers', this.apiUrl)));
+        this.imageServer = imageServers.find(server => server.id === 'main' && server.site_ids.includes(this.siteId));
```

Comment:
How often are these servers changing?
If they almost never change, they may be hardcoded.
If they change daily, users may have to restart the application to update these servers since this is done in the `Initialize` method.

Reply:
When do you think i should update them? Make a request at the beginning of `FetchPages` ?